### PR TITLE
feat: introduce flakybot

### PR DESCRIPTION
**Pull request by JustinBeckwith**
Things I think I need to do after this:
- [ ] Double check on the Cloud Run service name.  If a new one gets created, delete the old one.
- [ ] Double check cloud scheduler name
- [ ] Rename the GitHub App itself, and verify existing installations come along
- [ ] Do a scrub broadly through the org for references to the prior bot name
- [ ] Write a script to remove the previous github labels we were adding so folks don't get confused

@tbpg do I need to worry about a PubSub queue name or something here?

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
Failing test is a bug in the CI script.  It sees that the `buildcop` folder was deleted, and treats that like a change that requires testing.  This only comes up when we delete bots, so an admin override here is likely fine.

**Review by tbpg — CHANGES_REQUESTED** [review]
I don't _think_ there is anything to do for the Pub/Sub queue. I think it only depends on the Cloud Function name, which will be updated.

The list of things to do after this looks good. I left a comment about supporting both sets of labels for some amount of time. That would be another TODO related to the script to move from old labels to new.

File: packages/flakybot/__snapshots__/buildcop.js [hunk 1]
```diff
@@ -0,0 +1,299 @@
+exports['flakybot app testsFailed opens an issue when testsFailed 1'] = {
```
[the remark this answers is not in the document]
Comment (JustinBeckwith) [reply packages/flakybot/__snapshots__/buildcop.js:1]: yup

File: metrics/dashboards.sql [hunk 2]
```diff
@@ -98,7 +98,7 @@ SELECT * FROM (SELECT COUNT(id) as actions, month_start, 6.75 as minutes, 'build
     repo.name LIKE 'googleapis/%' OR
     repo.name LIKE 'GoogleCloudPlatform/%'
   ) AND
-  actor.login LIKE "build-cop-bot%" AND
```
[the remark this answers is not in the document]
Comment (JustinBeckwith) [reply metrics/dashboards.sql:101]: done

File: packages/flakybot/src/flakybot.ts [hunk 3]
```diff
@@ -33,9 +33,9 @@ type IssuesListForRepoResponseItem = components['schemas']['issue-simple'];
 type IssuesListCommentsResponseData = components['schemas']['issue-comment'][];
 type IssuesListForRepoResponseData = IssuesListForRepoResponseItem[];
 
-const ISSUE_LABEL = 'buildcop: issue';
-const FLAKY_LABEL = 'buildcop: flaky';
-const QUIET_LABEL = 'buildcop: quiet';
+const ISSUE_LABEL = 'flakybot: issue';
+const FLAKY_LABEL = 'flakybot: flaky';
+const QUIET_LABEL = 'flakybot: quiet';
```
[the remark this answers is not in the document]
Comment (JustinBeckwith) [reply packages/flakybot/src/flakybot.ts:38]: This is going to be a one day thing - I'd prefer to leave it out and fix over the weekend if I have to

**Review by tbpg — APPROVED** [review]
Thank you!

**Pull request merged**